test/production/react-18-streaming-ssr/index.test.ts

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs-extra'
+import { join } from 'path'
+import { createNext } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import { renderViaHTTP } from 'next-test-utils'
+
+describe('react-18-streaming-ssr in minimal mode', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files: {
+        'pages/index.server.js': `
+          export default function Page() { 
+            return <p>static streaming</p>
+          }
+        `,
+      },
+      nextConfig: {
+        experimental: {
+          reactRoot: true,
+          serverComponents: true,
+          runtime: 'nodejs',
+        },
+      },
+      dependencies: {
+        react: '18.0.0-rc.0',
+        'react-dom': '18.0.0-rc.0',
+      },
+      skipStart: true,
+    })
+
+    // Enable minimal mode
+    const baseServerPath = join(
+      next.testDir,
+      'node_modules/next/dist/server/base-server.js'
+    )
+    const minimalModeBaseServerCode = fs
+      .readFileSync(baseServerPath, { encoding: 'utf-8' })
+      .replace(/this.minimalMode = minimalMode/, 'this.minimalMode = true')
+    fs.writeFileSync(baseServerPath, minimalModeBaseServerCode)
```

Comment:
This seems very hacky (e.g. if we update this line of code internally, it will fail)... can we make it an environment variable (pass in via https://github.com/vercel/next.js/blob/389432048cf6973d50d9d795c8dae7da123d8e31/test/lib/next-modes/next-start.ts#L46-L51)?

Reply:
We also have this example here, can we do something similar?

https://github.com/vercel/next.js/blob/a92a5caec2996e22c526df4fdf465a3db5086431/test/integration/required-server-files-ssr-404/test/index.test.js#L52-L57